### PR TITLE
Handle no-op custom call in the emitter

### PR DESCRIPTION
**Pull request by jaro-sevcik**
The AllocateBuffer custom call (a.k.a. `kNopCustomCallTarget`) is no-op because the runtime allocates the buffer. Let us handle AllocateBuffer by emitting nothing and returning.

### CODE REVIEW

**Comment by jaro-sevcik** [comment]
Here is a link to a motivation/summary doc: https://docs.google.com/document/d/12LjzvR0foAMnkB7p7-EtiuwdhfmsRAsEWYFTGGiOs7U/edit?usp=sharing

**Comment by jaro-sevcik** [comment]
Let me know if you think this needs a test in XLA (this will be eventually tested via JAX because host-offloading in the remat-scan test introduces the "AllocateBuffer" op).

**Comment by cheshire** [comment]
And otherwise, HLO with this custom call would crash now? Then it would be possible to write a test?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: xla/service/gpu/tests/allocate_buffer_custom_call_test.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+limitations under the License.
+==============================================================================*/
+
+#include "xla/tests/hlo_test_base.h"
+
+namespace xla {
+namespace gpu {
+namespace {
+
+class AllocateBufferTest : public HloTestBase {};
```
[the remark this answers is not in the document]
Comment (jaro-sevcik) [reply xla/service/gpu/tests/allocate_buffer_custom_call_test.cc]: Done (even though the test is about the allocation aspect of the custom call, so I find the "nop" somewhat misleading)

**Review by cheshire — APPROVED** [review]

**Pull request closed without merging**